### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by damianbraun** [StyraHem/ShellyForHASS#713]
## Environment

- ShellyForHASS release with the issue: 1.0.1
- Last working ShellyForHASS release (if known): 1.0.1 (under homeassiastant 2022.8.7
- Home Assistant Core release with the issue: 2022.9.0 
- Operating environment (Home Assistant/Supervised/Docker/venv):  HomeAssistant

## Describe the bug

After 2022.9.0 update, light entities of shelly dimmer 2's became unavailable.

## Traceback/Error logs

Logger: homeassistant.setup
Source: setup.py:320
First occurred: 20:14:23 (3 occurrences)
Last logged: 20:21:03

Unable to prepare setup for platform shelly.light: Platform not found (cannot import name 'ATTR_WHITE_VALUE' from 'homeassistant.components.light' (/usr/src/homeassistant/homeassistant/components/light/__init__.py)).
```txt

```

**Comment by kingunlim** [comment]
second this

**Comment by emilio77** [comment]
third with same problem

**Comment by HFeenstra** [comment]
Second this too. I thought I was the only one with issue, until I just found this bug.
Was thinking towards 2022.9 update of HA

**Comment by Olausson92** [comment]
Same problem after I updated to HA 20220907-0 today. 

Logger: homeassistant.setup
Source: setup.py:320
First occurred: 21:42:46 (1 occurrences)
Last logged: 21:42:46

`Unable to prepare setup for platform shelly.light: Platform not found (cannot import name 'ATTR_WHITE_VALUE' from 'homeassistant.components.light' (/usr/src/homeassistant/homeassistant/components/light/__init__.py)).`

**Comment by mattlamb99** [comment]
Same issue after todays upgrade.

Logger: homeassistant.setup
Source: setup.py:320
First occurred: 8:33:12 AM (1 occurrences)
Last logged: 8:33:12 AM

Unable to prepare setup for platform shelly.light: Platform not found (cannot import name 'ATTR_WHITE_VALUE' from 'homeassistant.components.light' (/usr/src/homeassistant/homeassistant/components/light/__init__.py)).

**Comment by mattlamb99** [comment]
Found the issue cause:
`Remove white_value support from light ([@emontnemery](https://github.com/emontnemery) - [#76926](https://github.com/home-assistant/core/pull/76926)) ([light docs](https://www.home-assistant.io/integrations/light/)) ([group docs](https://www.home-assistant.io/integrations/group/)) ([flux docs](https://www.home-assistant.io/integrations/flux/)) ([switch_as_x docs](https://www.home-assistant.io/integrations/switch_as_x/))`

**Comment by HFeenstra** [comment]
> Found the issue cause:
> Remove white_value support from light ([@emontnemery](https://github.com/emontnemery) - [#76926](https://github.com/home-assistant/core/pull/76926)) ([light docs](https://www.home-assistant.io/integrations/light/)) ([group docs](https://www.home-assistant.io/integrations/group/)) ([flux docs](https://www.home-assistant.io/integrations/flux/)) ([switch_as_x docs](https://www.home-assistant.io/integrations/switch_as_x/))

Let’s hope someone can have a look at it. Hmm something was ready to get removed 1,5 years ago..?? Finally removed, I guess I missed the breaking changes

**Comment by mattlamb99** [comment]
Have made a very crude fix to at least get most of your lights working again - but this doesn't fix RGBW lights
https://github.com/StyraHem/ShellyForHASS/pull/714

**Comment by arturkoko** [comment]
Same issue

**Comment by antoweb** [comment]
same issue

**Comment by antoweb** [comment]
any solution or we are waiting for a new release?

**Comment by Moxser** [comment]
I have the same problem. 

Logger: homeassistant.setup
Source: setup.py:320
First occurred: 11:04:32 (1 occurrences)
Last logged: 11:04:32

Unable to prepare setup for platform shelly.light: Platform not found (cannot import name 'ATTR_WHITE_VALUE' from 'homeassistant.components.light' (/usr/src/homeassistant/homeassistant/components/light/__init__.py)).

**Comment by kingunlim** [comment]
> Have made a very crude fix to at least get most of your lights working again - but this doesn't fix RGBW lights
> #714

hey thx, how can i use your fix?

best regards

**Comment by piet1951** [comment]
Same problem, I restored a backup. What to do now ?

**Comment by antoweb** [comment]
> > Have made a very crude fix to at least get most of your lights working again - but this doesn't fix RGBW lights
> > #714
> 
> hey thx, how can i use your fix?
> 
> best regards

Just remove this lines from light.py
    ATTR_WHITE_VALUE,
    SUPPORT_BRIGHTNESS, SUPPORT_COLOR, SUPPORT_COLOR_TEMP, SUPPORT_EFFECT,
    SUPPORT_WHITE_VALUE,
-----
       if ATTR_WHITE_VALUE in kwargs:
            white_value = int(kwargs[ATTR_WHITE_VALUE])
            self._white_value = white_value

as speciifed in this url: https://github.com/StyraHem/ShellyForHASS/pull/714/commits/ca4b860f6985b3c3e845f74f93bfc9e661a7959e

and restart HA

**Comment by antoweb** [comment]
> 

You are welcome. But this is a workaround, we are waiting for a definitive solution

**Comment by hakana** [comment]
We working on a new release.

**Comment by Ragbak99** [comment]
After deleting those lines the error os gone. Thanx

**Comment by david-kalbermatten** [comment]
Duplicate of #706

**Comment by doug62** [comment]
+1 - Same problem

**Comment by dolenec** [comment]
Any timeline when the fix will be officially published?

**Comment by Friedrieck** [comment]
> ATTR_WHITE_VALUE,
> **SUPPORT_BRIGHTNESS, SUPPORT_COLOR, SUPPORT_COLOR_TEMP, SUPPORT_EFFECT,**
> SUPPORT_WHITE_VALUE,

To those wanting to correct it by themselves, keep the second line of this block above (as specified in the original fix: https://github.com/StyraHem/ShellyForHASS/commit/ca4b860f6985b3c3e845f74f93bfc9e661a7959e).

**Comment by convicte** [comment]
To simplify it even further, remove ONLY the red lines, and leave everything as is.
![image](https://user-images.githubusercontent.com/78666770/189325647-70026778-7221-4274-a3fc-04cf86f17f36.png)

Close the file and restart HA.

**Comment by lwYeo** [comment]
> > ATTR_WHITE_VALUE,
> > **SUPPORT_BRIGHTNESS, SUPPORT_COLOR, SUPPORT_COLOR_TEMP, SUPPORT_EFFECT,**
> > SUPPORT_WHITE_VALUE,
> 
> To those wanting to correct it by themselves, keep the second line of this block above (as specified in the original fix: [ca4b860](https://github.com/StyraHem/ShellyForHASS/commit/ca4b860f6985b3c3e845f74f93bfc9e661a7959e)).

This does not fix RGBW2 devices, anyone has a quick workaround for RGBW2?

**Comment by TohtoriR** [comment]
same issue here. I'm not in a hurry with the fix as restored a backup but would obviously like to have the new bells and whistles from the new core. When could this fix be expected roughly? I do have those rgbw bulbs so the workaround is probably not for me.

**Comment by tmjo** [comment]
Same issue, please fix. The "manual" fix referenced above fixes the issue for me temporarily (Dimmer2).

**Comment by dreerr** [comment]
> To simplify it even further, remove ONLY the red lines, and leave everything else as is. ![image](https://user-images.githubusercontent.com/78666770/189325647-70026778-7221-4274-a3fc-04cf86f17f36.png)
> 
> Close the file and restart HA.

I can also confirm that the temporary fix is working, I had issues with Shelly Bulb Duo and Shelly Dimmer

**Comment by pljda** [comment]
> > To simplify it even further, remove ONLY the red lines, and leave everything else as is. ![image](https://user-images.githubusercontent.com/78666770/189325647-70026778-7221-4274-a3fc-04cf86f17f36.png)
> > Close the file and restart HA.
> 
> I can also confirm that the temporary fix is working, I had issues with Shelly Bulb Duo and Shelly Dimmer

Unfortunately it did not worked for me :(
Hope Shelly will issue an update on this matter !

**Comment by bjeanes** [comment]
That diff is not complete, I think. You also have to delete the line which references `SUPPORT_WHITE_VALUE`. I have the Dimmer 2 and it did work for me.

**Comment by SamKr** [comment]
Same problem here

**Comment by vorreiter** [comment]
Same problem

**Comment by taanczos** [comment]
Same here :( my fishes now lack of sunrise and moonlight :P

**Comment by hakana** [comment]
I have created a new release 1.0.2. Now it is in pre-release/beta mode. Can you please test and I will change it to a full release shortly.

**Comment by hakana** [comment]
This fix removes the white_channel for RGBW2 in color mode. We looking for a solution for this.
I think the only solution is to add another entity for the white channel in parallel to the RGB entity. Does someone know a better solution? I'm very busy right now so any help is needed!

**Comment by taanczos** [comment]
Hey! Thank you. It works for my rgbw2 and shelly dimmers. But I dont use the white channel with the rgbw2 only the rgb.

**Comment by pljda** [comment]
> I have created a new release 1.0.2. Now it is in pre-release/beta mode. Can you please test and I will change it to a full release shortly.

Hi Hakana, I think that you made it. My Dimmer & Duo lights are back to life :) This is good news.
Thanks for your quick action !
philippe

**Comment by Dilergore** [comment]
> 

I think that solution should be okay. Anyway I do not think anyone is using White + RGB at the same time. Even on the UI I have a single slider for brightness today which I control differently: if 'white_value > 0' I change white_value else I control brightness. In addition I have several pre-defined colors for easy change and there is one which changes brightness to '1' and white_value to 255 (so this is the white setting basically).

I've been using MQTT so far for RGBW2 but even with that the issue is the same. You could use the default light template and more or less it works, but it is not perfect.

Another workable solution I could imagine is to have a virtual 'mode' switch which needs to be set either 'white' or 'RGB' and the brightness would control the white brightness in white mode and the RGB brightness in RGB mode.

**Comment by dolenec** [comment]
Maybe it is just time to migrate to native Shelly integration.. I thought that this integration would be up-to-date not that something is forgotten to implement and that author do not have time.. sorry - but this is probably something that author need to be hear..

If there is no time, no problem, but be honest and publish this and give project to someone else.. simply my humble opinion.

PS. this so called hotfix broke rgbw2 behaive as was before, so it is not hotfix but half way baked patch..

**Comment by SamKr** [comment]
@dolenec you know they're sharing this for free, developing it in their own spare time? It's not a product you've bought. The speed with which they deployed this hotfix (even though the auther mentioned he's very busy atm) is something to thank them for, not to get annoyed at :\

Perhaps you'd be better off buying something off the shelves?

**Comment by david-kalbermatten** [comment]
> Another workable solution I could imagine is to have a virtual 'mode' switch which needs to be set either 'white' or 'RGB' and the brightness would control the white brightness in white mode and the RGB brightness in RGB mode.

That's actually what the new `ColorMode.WHITE` is supposed to be used for. See: https://github.com/home-assistant/core/pull/51411 and https://developers.home-assistant.io/docs/core/entity/light/#color-modes

**Comment by fh-lipweeyeo** [comment]
> > Another workable solution I could imagine is to have a virtual 'mode' switch which needs to be set either 'white' or 'RGB' and the brightness would control the white brightness in white mode and the RGB brightness in RGB mode.
> 
> That's actually what the new `ColorMode.WHITE` is supposed to be used for. See: https://github.com/home-assistant/core/pull/51411 and https://developers.home-assistant.io/docs/core/entity/light/#color-modes
> 

Note that there is actually an enum called ColorMode.RGBW in the link.

**Comment by doug62** [comment]
FWIW - I purchase cheap LED strips from China, $8/15ft, I have about 15 RGBW2s, I DO USE RGB+W to change the color/temperature of the whites,  There is a way to control W+RGB, the WIZ integration does this, there is also a way to factor W into the RBG in some color spaces, I haven't reviewed the details but: 1) Folks do use RGB+W; 2) There is an existing pattern (WIZ)  And sorry - though I'm experienced C# dev I know/care nothing about Python so can't really contribute but THANKS to those who do!

**Comment by doug62** [comment]
Assuming that this broke RGBW (which I'm heavily dependent upon), it is probably time to move to the native integration - does anyone know of any reasons why/how to do this?  Is one "better" than the other?

**Comment by bigboban** [comment]
It's a trap! This integration has RGBW feature broken. And official HA integration does NOT support Shelly4pro. For me there is no fully functional integration for all shelly devices. Only way is use one or other integration and integrate unsupported device(s) some other way (REST, MQTT, ..).

**Comment by doug62** [comment]
FWIW - The core/shelly does have both color+white on RGBW2

**Comment by doug62** [comment]
@hakana - Thanks for your work, earlier you were asking for a pattern, i have confirmed that the core shelly rgbw2 works same as this used to flawlessly, I'm not a python dev, looking at their code at https://github.dev/home-assistant/core/blob/38b087d2209dbe26f90676ecebf90fc1b97836e3/homeassistant/components/shelly/light.py#L12

**Comment by hakana** [comment]
How is it working in core integration. Is there two entities, one for rgb and one for white channel? Can you take a screen shot?
________________________________

**Comment by doug62** [comment]
WIZ: (Color) 
![image](https://user-images.githubusercontent.com/5632034/190861567-29759be4-316c-4f14-b2a3-c8ff520af172.png)

WIZ Temperature:
![image](https://user-images.githubusercontent.com/5632034/190861592-4394767f-4d42-412b-bdb3-da5201e64d14.png)

LIFX: 
![image](https://user-images.githubusercontent.com/5632034/190861514-4aa4064f-e39a-4e1c-902a-26d6c3ad9cbd.png)

**Comment by david-kalbermatten** [comment]
> How is it working in core integration. Is there two entities, one for rgb and one for white channel? Can you take a screen shot?

The core integration offers white brightness and color brightness but the overall brightness slider only affects the RGB component.
![image](https://user-images.githubusercontent.com/45201085/190868284-3d14a786-062e-4b42-a260-3e3451c72976.png)

**Comment by Dilergore** [comment]
I have a more or less working setup with MQTT default light schema, if you want to use it just change the "<deviceId>" to proper value:
```yaml
- name: "ledstrip"
      qos: 2
      rgbw_command_topic: "shellies/shellyrgbw2-<deviceId>/color/0/set"
      rgbw_state_topic: "shellies/shellyrgbw2-<deviceId>/color/0/status"
      rgbw_command_template: >
        {"red": {{ red }}, "green": {{ green }}, "blue": {{ blue }}, "white": {{ white }} }
      rgbw_value_template: >
        { {{value_json.red}},{{value_json.green}},{{value_json.blue}},{{value_json.white}} }
      brightness_scale: 100
      brightness_command_topic: "shellies/shellyrgbw2-<deviceId>/color/0/set"
      brightness_state_topic: "shellies/shellyrgbw2-<deviceId>/color/0/status"
      brightness_command_template: >
        {"gain": {{ value }} }
      brightness_value_template: >
        {  {{value_json.gain}} }
      command_topic: "shellies/shellyrgbw2-<deviceId>/color/0/command"
      state_topic: "shellies/shellyrgbw2-<deviceId>/color/0/status"
      payload_on: "on"
      payload_off: "off"
      state_value_template: >
        {%- if value_json.ison -%}
          on
        {%- else -%}
          off
        {%- endif -%}
      effect_list:
        - 0
        - 1
        - 2
        - 3
        - 4
        - 5
        - 6
      effect_command_topic: "shellies/shellyrgbw2-<deviceId>/color/0/set"
      effect_state_topic: "shellies/shellyrgbw2-<deviceId>/color/0/status"
      effect_command_template: >
        {"effect": {{ value }} }
      effect_value_template: >
        {{ value_json.effect }}
```

The way this works is the following:
- Brightness controls RGB only
- On the more-info dialog there is a "Color Brightness" which is basically just changing RGB values with a slider between 0-255 (more or less useless if ask me)
- On the more-info dialog there is a "White Brightness" slider which can be used to control the White channel.

The problem with this in general is that while the normal more-info dialog has an option for "White Brightness", you cannot add it on the UI in any way as a slider (or I haven't yet found an option).
